### PR TITLE
Add bookoo scales support.

### PR DESCRIPTION
**Pull request by Zer0-bit**
**Purpose of this PR is to add support for the bookoo scales.**
image

- Added bookoo implementation keeping to the library format spec
- Moved some common code to a common class

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -12,7 +12,8 @@
 framework = arduino
 platform = https://github.com/platformio/platform-espressif32.git
 build_flags =
-	-std=gnu++17
+	-std=gnu++2a
```
Comment (Zer0-bit) [inline platformio.ini:15]: Added this so it builds on Windows (needs to be checked is still builds on other systems)

File: platformio.ini [hunk 2]
```diff
@@ -12,7 +12,8 @@
...
+	-frtti
```
Comment (Zer0-bit) [inline platformio.ini:16]: Added this so it builds on Windows as it seems Arduino framework was not compiled with this flag set so it errors out on dynamic_cast.

File: src/scales/bookoo.cpp [hunk 3]
```diff
@@ -0,0 +1,250 @@
+#include "bookoo.h"
+#include "remote_scales_plugin_registry.h"
+
+/*
+Handle protocol according to the spec found at
+https://github.com/BooKooCode/OpenSource/blob/main/bookoo_mini_scale/protocols.md
+*/
+
+enum class BookooHeader : uint8_t {
```
Comment (kstam) [inline src/scales/bookoo.cpp]: looks redundant?

Comment (Zer0-bit) [reply src/scales/bookoo.cpp]: Yup spotted that right, i left hem there as was thinking they are needed originally but ended up unnecessary and i just forgot to slay lol

**Review by kstam — APPROVED** [review]

**Pull request merged**